src/actions/google/drive/google_drive.ts

```diff
@@ -105,59 +106,78 @@ export class GoogleDriveAction extends Hub.OAuthAction {
+            // drive.files.list() options
+            const options: any = {
+              fields: "files(id,name,parents),nextPageToken",
+              orderBy: "recency desc",
+              pageSize: 1000,
+              q: `mimeType='application/vnd.google-apps.folder' and trashed=false`,
+              spaces: "drive",
+            }
+            if (request.formParams.drive !== undefined && request.formParams.drive !== "mydrive") {
+              options.driveId = request.formParams.drive
+              options.includeItemsFromAllDrives = true
+              options.supportsAllDrives = true
+              options.corpora = "drive"
+            } else {
+              options.corpora = "user"
+            }
 
-            // When a `nextPageToken` exists, recursively call this function to get the next page.
-            if (response.data.nextPageToken) {
-              const pageOptions = { ...options }
-              pageOptions.pageToken = response.data.nextPageToken
-              return pagedFileList(mergedFiles, await drive.files.list(pageOptions))
+            async function pagedFileList(
+                accumulatedFiles: drive_v3.Schema$File[],
+                response: GaxiosResponse<drive_v3.Schema$FileList>): Promise<drive_v3.Schema$File[]> {
+              const mergedFiles = accumulatedFiles.concat(response.data.files!)
+
+              // When a `nextPageToken` exists, recursively call this function to get the next page.
+              if (response.data.nextPageToken) {
+                const pageOptions = { ...options }
+                pageOptions.pageToken = response.data.nextPageToken
+                return pagedFileList(mergedFiles, await drive.files.list(pageOptions))
+              }
+              return mergedFiles
             }
-            return mergedFiles
+            const paginatedFiles = await pagedFileList([], await drive.files.list(options))
+            const folders = paginatedFiles.filter((folder) => (
+                    !(folder.id === undefined) && !(folder.name === undefined)))
+                .map((folder) => ({name: folder.id!, label: folder.name!}))
+            folders.unshift({name: "root", label: "Drive Root"})
+
+            form.fields.push({
+              description: "Google Drive folder where your file will be saved",
+              label: "Select folder to save file",
+              name: "folder",
+              options: folders,
+              default: folders[0].name,
+              required: true,
+              type: "select",
+            })
+            form.state = new Hub.ActionState()
+            form.state.data = JSON.stringify({tokens: stateJson.tokens, redirect: stateJson.redirect})
+          // We did not fetch the folder, offer to fetch or to enter a folderid
+          } else {
+            form.fields.push({
+              description: "Add url of directory that will receive the data.",
+              label: "Folder ID",
```

Comment:
Can we change this to `Folder URL` I can see the case where the user doesn't read the description and sees only `Folder ID` and it would fail because we can't handle just the folder id.